cranelift/codegen/src/machinst/vcode.rs

```diff
@@ -877,6 +879,26 @@ impl<I: VCodeInst> VCode<I> {
                 last_offset = Some(cur_offset);
             }
 
+            let lb = self.block_order.lowered_order()[block.index()];
+            let b = if let Some(b) = lb.orig_block() {
```

Comment:
This logic feels a little bit brittle to me. If I understand correctly, it is determining what the "real" CLIF-level block for this target is meant to be, even if we have inserted an edge block, so that we can tell if any indirect branch might reach it; is that right?

If so, I think what we really should have is a set of *lowered* block indices reachable via an indirect branch, computed by the lowering-order code. We can compute this when we traverse the lowered-block's successors if we know that we're visiting out-edges of an indirect branch, I think.